test/integration/common/StudioActions.kt

```diff
@@ -105,26 +102,31 @@ object Utils {
         beforeRetry: (() -> Unit) = {},
         successCondition: () -> Boolean
     ) {
+        var success = false
         val deadline = System.currentTimeMillis() + 10_000
-        while (!successCondition() && System.currentTimeMillis() < deadline) {
-            beforeRetry()
-            delayAndRecompose(context, 500)
+        while (!success && System.currentTimeMillis() < deadline) {
+            try {
+                if (successCondition()) {
+                    success = true
+                } else {
+                    beforeRetry()
+                    delayAndRecompose(context, 500)
+                }
+            } catch (e: Exception) {
+                LOGGER.error(e.stackTraceToString())
```

Comment:
We log any errors that do occur - though they may be errors thrown due to intermediate states. This doesn't effect the outcome of the program in the successful series of events and prints useful information if the test fails due to a faulty `successCondition`.